geomstats/geometry/spd_matrices.py

```diff
@@ -38,36 +38,20 @@ def vector_from_symmetric_matrix(self, mat):
         assert gs.all(self.embedding_manifold.is_symmetric(mat))
         mat = self.embedding_manifold.make_symmetric(mat)
 
-        _, mat_dim, _ = mat.shape
-        vec_dim = int(mat_dim * (mat_dim + 1) / 2)
-        vec = gs.zeros(vec_dim)
-
-        idx = 0
-        for i in range(mat_dim):
-            for j in range(i + 1):
-                if i == j:
-                    vec[idx] = mat[j, j]
-                else:
-                    vec[idx] = mat[j, i]
-                idx += 1
-
-        return vec
+        _, dim, _ = mat.shape
+        i, j = gs.tril_indices(dim)
+        return mat[:, i, j]
```

Comment:
🎉

Reply:
Yes I updated the decorator, but it was not tested at all before because the test's name did not start by test...